designer/fitting/dwipi.py

```diff
@@ -1655,175 +1656,294 @@ def optimPipeline(self, savePath=os.getcwd()):
+                                          cartIdx[2, i]]) - \
+                                np.mean(patchImg)) / \
+                    self.Img[cartIdx[0,i], cartIdx[1,i], cartIdx[2,i]]
+            if pdiff > self.Threshold:
+                # Mark outlier mask voxel as true
+                self.OutlierMask[cartIdx[0, i],
+                                 cartIdx[1, i],
+                                 cartIdx[2, i]] = True
+        # Create (3 x nvox) cartesian coordinate representation of
+        # image within a brainmask
+        self.CartIdx = np.array(np.where(self.OutlierMask > 0))
+        nvox = self.CartIdx.shape[1]
+        self.PatchIdx = np.zeros(nvox)
+        inputs = tqdm(range(nvox),
+                      desc='Locating Replacements   ',
+                      unit='vox',
+                      ncols=tqdmWidth)
+        cntSkip = 0
+        for i in inputs:
+            # Index beginning and ending of patch
+            Ib = self.CartIdx[0, i] - d2move
+            Ie = self.CartIdx[0, i] + d2move
+            Jb = self.CartIdx[1, i] - d2move
+            Je = self.CartIdx[1, i] + d2move
+            Kb = self.CartIdx[2, i] - d2move
+            Ke = self.CartIdx[2, i] + d2move
+            if self.Connectivity == 'all':
+                # Remove 14th (centroid) element
+                patchViol = np.delete(np.ravel(
+                    self.OutlierMask[Ib:Ie+1, Jb:Je+1, Kb:Ke+1]), 13)
+                patchImg = np.delete(np.ravel(
+                    self.Img[Ib:Ie + 1, Jb:Je + 1,Kb:Ke + 1]),
+                                     13)
+                connLimit = 26
+            elif self.Connectivity == 'face':
+                patchViol = self.OutlierMask[[Ib, Ie],
+                                             self.CartIdx[1, i],
+                                             self.CartIdx[2, i]]
+                patchViol = np.hstack((patchViol,
+                                       self.OutlierMask[self.CartIdx[0, i],
+                                                        [Jb, Je],
+                                                        self.CartIdx[2, i]]))
+                patchViol = np.hstack((patchViol,
+                                       self.OutlierMask[self.CartIdx[0, i],
+                                                        self.CartIdx[1, i],
+                                                        [Kb, Ke]]))
+                patchImg = self.Img[[Ib, Ie], self.CartIdx[1, i], self.CartIdx[2, i]]
+                patchImg = np.hstack((patchImg,
+                                      self.Img[self.CartIdx[0, i], [Jb, Je],
+                                               self.CartIdx[2, i]]))
+                patchImg = np.hstack((patchImg, self.Img[self.CartIdx[0, i],
+                                                         self.CartIdx[1, i],
+                                                         [Kb, Ke]]))
+                nVoil = np.sum(patchViol)
```

Comment:
Do you mean `nViol` here?

Reply:
It's meant to be `nVoil`, but I've replaced `nVoil` to `nViol` for consistency.